toolchain/check/diagnostic_emitter.cpp

```diff
@@ -71,40 +62,31 @@ auto DiagnosticEmitter::ConvertLocImpl(SemIR::LocId loc_id,
   return ConvertLocInFile(final_node_id, token_only, context_fn);
 }
 
-auto DiagnosticEmitter::TryConvertClangDiagnosticLoc(SemIR::LocId loc_id) const
-    -> std::optional<Diagnostics::ConvertedLoc> {
-  if (loc_id.kind() != SemIR::LocId::Kind::ImportIRInstId) {
-    return std::nullopt;
-  }
-
-  SemIR::ImportIRInst import_ir_inst =
-      sem_ir_->import_ir_insts().Get(loc_id.import_ir_inst_id());
-
-  if (import_ir_inst.ir_id() != SemIR::ImportIRId::Cpp) {
-    return std::nullopt;
-  }
-
-  clang::SourceLocation clang_loc =
-      sem_ir_->clang_source_locs().Get(import_ir_inst.clang_source_loc_id());
-
-  CARBON_CHECK(sem_ir_->cpp_ast());
-  clang::PresumedLoc presumed_loc =
-      sem_ir_->cpp_ast()->getSourceManager().getPresumedLoc(clang_loc);
-
-  return Diagnostics::ConvertedLoc{
-      .loc = {.filename = presumed_loc.getFilename(),
-              .line_number = static_cast<int32_t>(presumed_loc.getLine())},
-      // TODO: Set `last_byte_offset` based on the `import Cpp` location.
-      .last_byte_offset = 0};
-}
-
 auto DiagnosticEmitter::ConvertLocInFile(SemIR::AbsoluteNodeId absolute_node_id,
                                          bool token_only,
                                          ContextFnT /*context_fn*/) const
     -> Diagnostics::ConvertedLoc {
+  if (absolute_node_id.check_ir_id() == SemIR::CheckIRId::Cpp) {
+    // Special handling of Clang source locations.
+    // TODO: Add `InCppImport` here using `context_fn` similar to `InImport`
+    // instead of a note.
```

Comment:
Really the format should be identical to what `InImport` is already doing, so I think the goal should be to eliminate `InCppImport`. So I'll suggest a different TODO:

```suggestion
    // TODO: Refactor to add an `InImport` pointing at the `Cpp` import, and eliminate `InCppImport`.
```

Reply:
To be sure, this is another case where I'd say code sharing would be appropriate.

Reply:
Done.